Sources/ProcedureKit/Any.swift

```diff
@@ -6,50 +6,108 @@
 
 import Dispatch
 
-class AnyProcedureBox_<Input, Output>: GroupProcedure, InputProcedure, OutputProcedure {
-    var input: Pending<Input> = .pending
-    var output: Pending<ProcedureResult<Output>> = .pending
-}
+struct ValueBox<Value> {
+
+    typealias Getter = (Void) -> Value
+    typealias Setter = (Value) -> Void
 
-class AnyProcedureBox<Base: Procedure>: AnyProcedureBox_<Base.Input, Base.Output> where Base: InputProcedure & OutputProcedure {
+    private let setter: Setter
+    private let getter: Getter
 
-    private var base: Base
+    var value: Value {
+        get {
+            return getter()
+        }
+        set {
+            setter(newValue)
+        }
+    }
+
+    init(getter: @escaping Getter, setter: @escaping Setter) {
+        self.getter = getter
+        self.setter = setter
+    }
+}
+
+enum ProcedureOutputBoxCreator {
+    static func outputBox<P: OutputProcedure>(for procedure: P) -> ValueBox<Pending<ProcedureResult<P.Output>>> {
+        let getter = { return procedure.output }
```

Comment:
It should not change procedure lifecycle. Group procedure holds strong reference to child procedure anyway. But `unowned` also will work. So that's more about code conventions.